synthesizer/process/src/cost.rs

```diff
@@ -53,10 +59,11 @@ pub fn deployment_cost<N: Network>(deployment: &Deployment<N>) -> Result<(u64, (
     // Compute the total cost in microcredits.
     let total_cost = storage_cost
         .checked_add(synthesis_cost)
+        .and_then(|x| x.checked_add(constructor_cost))
         .and_then(|x| x.checked_add(namespace_cost))
         .ok_or(anyhow!("The total cost computation overflowed for a deployment"))?;
 
-    Ok((total_cost, (storage_cost, synthesis_cost, namespace_cost)))
+    Ok((total_cost, (storage_cost, synthesis_cost, constructor_cost, namespace_cost)))
```

Comment:
After https://github.com/ProvableHQ/snarkOS/pull/3471 is merged, you'll have to make a snarkOS PR to support the interface update.

In case we wait with this PR being merged until audits are completed, maybe a single snarkOS draft PR for all the program upgradability stuff is easier.

Reply:
Can you draft a snarkOS PR? :)

Reply:
https://github.com/ProvableHQ/snarkOS/pull/3587